feat(shared/codeHighlight): add additional syntax highlight languages

jimeh opened the pull request:
## 🧢 Changes

Add syntax highlighting for additional languages in diff view. Specifically added Go, YAML, TOML, Lisp, Lua, and Swift. I have tested that each language does correctly syntax highlight files via the dev build.

This is my first change to Gitbutler, and some of the first TypeScript I touch, so apologies if anything is off or weird. I'd be more than happy to implement any changes. I couldn't spot any relevant tests that might have needed updates, but it is very possible I'm blind.

## ☕️ Reasoning

I primarily wanted support for Go, and also added a few more languages I commonly use, along with Swift to resolve #2684 as well while I was in the area.

## 🎫 Affected issues

Fixes: #2684

## Screenshots

<details>
  <summary>Dockerfile</summary>

![Screen-Capture-2025-04-06-02-27-30](https://github.com/user-attachments/assets/3c2a634d-caed-4fe1-badc-03deb5561c95)

</details>

<details>
  <summary>Go</summary>

![Screen-Capture-2025-04-01-02-39-03](https://github.com/user-attachments/assets/0a36e5e0-52e8-481f-8d93-3e033c201f1a)

</details>

<details>
  <summary>Jinja2</summary>

![Screen-Capture-2025-04-06-02-28-01](https://github.com/user-attachments/assets/9f825978-b121-4a67-917d-13e13fda7b93)

</details>

<details>
  <summary>Lisp</summary>

![Screen-Capture-2025-04-01-02-40-22](https://github.com/user-attachments/assets/da7507a7-f769-4a70-81b1-2d4f8e9dab23)

</details>

<details>
  <summary>Lua</summary>

![Screen-Capture-2025-04-01-02-39-16](https://github.com/user-attachments/assets/fffbf9fc-5e40-4baf-9f0d-90ba4702552f)

</details>

<details>
  <summary>sh/bash/zsh</summary>

![Screen-Capture-2025-04-06-02-27-45](https://github.com/user-attachments/assets/f3357c11-6b03-4f5f-977f-18cc572663c9)

</details>

<details>
  <summary>Swift</summary>

![Screen-Capture-2025-04-01-02-39-34](https://github.com/user-attachments/assets/64912e10-804e-4c42-b207-d959ba17bbfa)

</details>

<details>
  <summary>TOML</summary>

![Screen-Capture-2025-04-01-02-38-49](https://github.com/user-attachments/assets/faf16b1b-6631-4125-ac59-26143640dd41)

</details>

<details>
  <summary>YAML</summary>

![Screen-Capture-2025-04-01-02-38-36](https://github.com/user-attachments/assets/29d9d736-335e-4b6e-b274-98dcfd76659b)

</details>

jimeh committed:
feat(shared/codeHighlight): add Go syntax highlighting support

Add CodeMirror's Go language parser to the codeHighlight module, 
allowing syntax highlighting for .go files and code blocks.

The changes:
- Install `@codemirror/lang-go` package as a dependency
- Add Go file extension detection in parserFromFilename
- Reorganize package.json dependencies alphabetically
remove golang extension check

jimeh committed:
feat(shared/codeHighlight): add Lua syntax highlighting support

Add Lua language to the code highlighting support by importing the Lua 
module from CodeMirror legacy modes and registering it in the 
parserFromFilename function.

jimeh committed:
feat(shared/codeHighlight): add YAML syntax highlighting support

Add `@codemirror/lang-yaml` dependency and configure it in the code 
highlight module to enable syntax highlighting for YAML files. Both 
.yml and .yaml file extensions are now supported with proper syntax 
highlighting.

jimeh committed:
feat(shared/codeHighlight): add TOML syntax highlighting support

Add support for TOML file syntax highlighting by importing the TOML 
language mode from CodeMirror legacy modules and configuring the 
parser to recognize .toml file extensions. This extends our code
highlighting capabilities to support TOML configuration files.

jimeh committed:
feat(shared/codeHighlight): add Swift syntax highlighting support

Add support for Swift language highlighting in the code highlighter
component. This enables proper syntax highlighting for Swift code by
importing the swift language module from @codemirror/legacy-modes and
adding a parser definition for files with the 'swift' extension.

jimeh committed:
feat(shared/codeHighlight): add Lisp syntax highlighting support

Add support for Common Lisp language highlighting in code snippets.
This includes importing the commonLisp module from CodeMirror's
legacy modes and configuring the parser to recognize .lisp, .cl, 
and .el file extensions. The implementation also handles Emacs Lisp
files which share the .el extension.

Byron reviewed (commented):
Thanks so much for contributing, and the screenshots are a great idea, too!

Since a lot of care went into the individual commits, a Butler Review would be great to have for review as well.

In any case, @estib-vega will know what to do here.

jimeh committed:
feat(shared/codeHighlight): add shell script syntax highlighting support

Add support for syntax highlighting of shell scripts (.sh, .bash, .zsh)

jimeh committed:
feat(shared/codeHighlight): add Dockerfile syntax highlighting support

This change enhances code highlighting functionality by adding support 
for Dockerfile syntax.

It specifically handles files named exactly `Dockerfile` (case-sensitive), begins with exactly `Dockerfile.` (case-sensitive), or has file extension `dockerfile`  (case-insensitive).

jimeh committed:
feat(shared/codeHighlight): Add Jinja2 syntax highlighting support

jimeh commented:
I've just rebased on top of latest changes from `master`, and also added syntax highlighting for shell scripts (sh, bash and zsh), Dockerfiles, and jinja2 template files.

With the shell scripts, I did also have a look if I could check any shabang comments at the beginning of the files to determine the language, but that was a no-go when I realized that the `create` function is called for each individual line, rather than each file as a whole.

But I guess that makes sense as we're operating on diffs rather than whole files. Though I assume things could be optimized to run once on a per-file basis rather than one per line. But that definitely falls outside of my experience level with TypeScript ^_^

Byron commented:
Thanks again - even better!

Indeed, I'd also want to try and get this PR merged as is, but am interested in learning what would need to be done to be able to fetch language information from shebangs, or even markdown codeblocks.
On top of that, it would be interesting if such a system could be tuned to allow highlighting markdown blocks in Rust files correctly.

I also realize that @krlvi was editing this (old) file more recently.

krlvi reviewed (approved):
Very nice contribution, thank you very much @jimeh!
